File: tests/runtime/smoke/host.ts
// Flags: --compat --map testwasi=./helpers.js,imports=./host.js --base64-cutoff=2500
function assert(x: boolean, msg: string) {
  if (!x)
    throw new Error(msg);
}

let hit = false;

export function thunk () {
  hit = true;
}

async function run() {
  const wasm = await import('./smoke.js');

  await wasm.$init;

  wasm.thunk();
  assert(hit, "import not called");
}

// Async cycle handling
setTimeout(run);

File: tests/runtime/smoke/host.py
from smoke import Smoke, SmokeImports
from helpers import TestWasi
import wasmtime

class MyImports:
    def thunk(self) -> None:
        self.hit = True

def run() -> None:
    store = wasmtime.Store()

    imports = MyImports()
    wasm = Smoke(store, SmokeImports(imports, TestWasi()))

    wasm.thunk(store)
    assert(imports.hit)

if __name__ == '__main__':
    run()
